certbot-apache/certbot_apache/configurator.py

```diff
@@ -1377,7 +1377,20 @@ def enhance(self, domain, enhancement, options=None):
             raise errors.PluginError(
                 "Unsupported enhancement: {0}".format(enhancement))
         try:
-            func(self.choose_vhost(domain), options)
+            # Temp value tells choose_vhost to not to create a new SSL vhost if
+            # one is not found.
+            vhost = self.choose_vhost(domain, temp=True)
+            if not vhost.ssl:
+                msg_tmpl = ("Certbot was not able to find SSL VirtualHost for a "
+                       "domain {0} for enabling enhancement \"{1}\". The requested "
+                       "enhancement was not configured.")
+                if options:
+                    msg = msg_tmpl.format(domain, options)
```

Comment:
I think whether `options` is a better description than `enhancement` depends on the enhancement type. If it's HSTS, `options` is better, if it's OCSP stapling, `enhancement` is better. What if when `options` is set, we put both `enhancement` and `options` in the error message?

Reply:
This sounds good. For the current enhancements the case was that options was only used for `ensure-http-header` enhancement, where `options` would hold the name of the actual header being added. Printing them both is probably the most informative way to handle this.